<launch>
  <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="launch_prefix" default="" />

  <!-- configure camera input -->
  <arg name="camera_name" default="/camera/camera/color" />
  <arg name="image_topic" default="image_raw" />
  <arg name="queue_size" default="1" />

  <node_container pkg="rclcpp_components" exec="component_container" name="apriltag_ros_continuous_detector_container" namespace="">
    <composable_node pkg="apriltag_ros" plugin="apriltag_ros::ContinuousDetector" name="apriltag_ros_continuous_detector_node" namespace="">
        <!-- Remap topics from those used in code to those on the ROS network -->
        <remap from="image_rect" to="$(var camera_name)/$(var image_topic)" />
        <remap from="camera_info" to="$(var camera_name)/camera_info" />

        <param name="publish_tag_detections_image" type="bool" value="true" /><!-- default: false -->
        <param name="queue_size" type="int" value="$(var queue_size)" />

        <!-- load parameters (incl. tag family, tags, etc.) -->
        <param from="$(find-pkg-share apriltag_ros)/config/tags.yaml"/>
        <param from="$(find-pkg-share apriltag_ros)/config/settings.yaml"/>
    </composable_node>
  </node_container>
</launch>

<!--
  <!- apriltag_ros continuous detection node
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" output="screen" launch-prefix="$(var launch_prefix)">
    <!- Remap topics from those used in code to those on the ROS network
    <remap from="image_rect" to="$(var camera_name)/$(var image_topic)" />
    <remap from="camera_info" to="$(var camera_name)/camera_info" />

    <param name="publish_tag_detections_image" type="bool" value="true" /><!- default: false ->
    <param name="queue_size" type="int" value="$(var queue_size)" />

    <!- load parameters (incl. tag family, tags, etc.) 
    <param from="$(find-pkg-share apriltag_ros)/config/tags.yaml"/>
    <param from="$(find-pkg-share apriltag_ros)/config/settings.yaml"/>
  </node>
-->
